src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -290,7 +283,8 @@ public void postInject() {
         this.iotJobsClientWrapper = iotJobsClientFactory.getIotJobsClientWrapper(connection);
 
         deviceConfiguration.onAnyChange((what, node) -> {
-            if (node != null && what.equals(WhatHappened.childChanged) && relevantNodeChanged(node)) {
+            if (node != null && what.equals(WhatHappened.childChanged)
```

Comment:
nit: it's a better habit to do `WhatHappened.childChanged.equals(what)` so that in case `what` is null we won't get null pointer exception. Same for line 176 in FSS.

Reply:
Changed in both places